### PR TITLE
[tests-only][full-ci] make user enable disable feature use graphapi

### PR DESCRIPTION
**Pull request by S-Panta**
## Description
The steps where admin disable any user issue were marked failed as it uses OCS API. At present, this task couldn't be done using ocs API as the curl request to that endpoint results in an error. This PR redacted that step so that the feature now uses graph API and the Scenario gets passed and they were removed from the expected failure file. 

## Related Issue
- https://github.com/owncloud/QA/issues/816

## How Has This Been Tested?

- locally
- ci

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/coreApiWebdavOperations/refuseAccess.feature [hunk 1]
```diff
@@ -27,7 +27,7 @@ Feature: refuse access
     Given using <dav-path-version> DAV path
     And user "Alice" has been created with default attributes and without skeleton files
     And user "Alice" has uploaded file "filesForUpload/textfile.txt" to "textfile0.txt"
-    And user "Alice" has been disabled
```
Comment (S-Panta) [inline tests/acceptance/features/coreApiWebdavOperations/refuseAccess.feature:30]: This step is also used in other scenarios and the step is failing and might be is in the expected to failure file somewhere. So shall we replace this step with new step `And the user "Admin" has disabled user "Alice" using the Graph API` and coreect the other scenarios as well ?
CC @phil-davis @saw-jan @SwikritiT

Comment (SwikritiT) [reply tests/acceptance/features/coreApiWebdavOperations/refuseAccess.feature:30]: Note we also have these issue for this https://github.com/owncloud/ocis/issues/6729 https://github.com/owncloud/ocis/issues/1420

I think we should refactor them all but maybe in separate PR's ?

File: tests/acceptance/features/coreApiWebdavOperations/refuseAccess.feature [hunk 2]
```diff
@@ -27,7 +27,7 @@ Feature: refuse access
...
+    And the user "Admin" has disabled user "Alice" using the Graph API
```
Comment (SwikritiT) [inline tests/acceptance/features/coreApiWebdavOperations/refuseAccess.feature:30]: Thi should be fine I don't think we need to mention the `using the Graph API` here. The stap should use the graph API internally? I don't think there's any other api that can be used to `enable/disable` user apart from graph?
```suggestion
    And the user "Admin" has disabled user "Alice"
```

Comment (S-Panta) [reply tests/acceptance/features/coreApiWebdavOperations/refuseAccess.feature:30]: There are ten other steps where `using the Graph API` is used. Should I refactor them in this PR along with the step definition code?

Comment (SwikritiT) [reply tests/acceptance/features/coreApiWebdavOperations/refuseAccess.feature:30]: No, do that in the next PR.

Comment (saw-jan) [reply tests/acceptance/features/coreApiWebdavOperations/refuseAccess.feature:30]: Yeas, let's merge this

**Review by saw-jan — APPROVED** [review]

**Review by saw-jan — CHANGES_REQUESTED** [review]
this test runs in reva. so need to find better way

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 3]
```diff
@@ -5402,20 +5407,21 @@ public function rememberGroupsThatExistAtTheStartOfTheScenario():void {
...
 			. "/ocs/v$this->ocsApiVersion.php/cloud/users/$actualOtherUser/$action";
-		$this->response = HttpRequestHelper::put(
+		$response = HttpRequestHelper::put(
 			$fullUrl,
 			$this->getStepLineRef(),
 			$actualUser,
 			$actualPassword
 		);
 		$this->pushToLastStatusCodesArrays();
+		return $response;
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/Provisioning.php:5424]:
```suggestion
		return HttpRequestHelper::put(
			$fullUrl,
			$this->getStepLineRef(),
			$actualUser,
			$actualPassword
		);
```

Comment (saw-jan) [reply tests/acceptance/features/bootstrap/Provisioning.php:5424]: Also, make sure that the response is stored wherever `disableOrEnableUser` is used in the When step

File: tests/acceptance/features/coreApiShareCreateSpecialToShares2/createShareWithDisabledUser.feature [hunk 4]
```diff
@@ -23,6 +22,4 @@ Feature: share resources with a disabled user
     And user "Brian" has been created with default attributes and without skeleton files
     And user "Alice" has been disabled
     When user "Alice" shares file "textfile0.txt" with user "Brian" using the sharing API
-    Then the OCS status code should be "997"
-    #And the OCS status code should be "401"
```
Comment (SagarGi) [inline tests/acceptance/features/coreApiShareCreateSpecialToShares2/createShareWithDisabledUser.feature:27]: this steps are also in expected to failures. Updating line numbers is required.

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 5]
```diff
@@ -3825,9 +3826,13 @@ public function theAdministratorDisablesTheFollowingUsersUsingTheProvisioningApi
...
 		$user = $this->getActualUsername($user);
-		$this->disableOrEnableUser($this->getAdminUsername(), $user, 'disable');
-		$this->theHTTPStatusCodeShouldBeSuccess();
-		$this->ocsContext->assertOCSResponseIndicatesSuccess();
+		if (OcisHelper::isTestingWithGraphApi()) {
+			$response = $this->graphContext->editUserUsingTheGraphApi($this->getAdminUsername(), $user, null, null, null, null, false);
+		} else {
+			$response = $this->disableOrEnableUser($this->getAdminUsername(), $user, 'disable');
+		}
+		$this->setResponse($response);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/Provisioning.php:3834]:
```suggestion
```

File: tests/acceptance/features/coreApiShareCreateSpecialToShares2/createShareWithDisabledUser.feature (hunk 4 above)
Comment (S-Panta) [reply tests/acceptance/features/coreApiShareCreateSpecialToShares2/createShareWithDisabledUser.feature:27]: These steps are passed and thus removed from expected failure file

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 6]
```diff
@@ -24,6 +24,7 @@
 use GuzzleHttp\Exception\GuzzleException;
 use Psr\Http\Message\ResponseInterface;
 use PHPUnit\Framework\Assert;
+use TestHelpers\GraphHelper;
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/Provisioning.php:27]:
```suggestion
```

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 7]
```diff
@@ -3825,9 +3826,15 @@ public function theAdministratorDisablesTheFollowingUsersUsingTheProvisioningApi
...
-		$this->ocsContext->assertOCSResponseIndicatesSuccess();
+		if (OcisHelper::isTestingWithGraphApi()) {
+			$response = $this->graphContext->editUserUsingTheGraphApi($this->getAdminUsername(), $user, null, null, null, null, false);
+		} else {
+			$response = $this->disableOrEnableUser($this->getAdminUsername(), $user, 'disable');
+		}
+		Assert::assertEquals(
+			200,
+			$response->getStatusCode()
+		);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/Provisioning.php:3837]: Please, add some nice error message there
```suggestion
		Assert::assertEquals(
			200,
			$response->getStatusCode(),
			"nice error message with response body"
		);
```

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 8]
```diff
@@ -3825,9 +3825,17 @@ public function theAdministratorDisablesTheFollowingUsersUsingTheProvisioningApi
...
+		if (OcisHelper::isTestingWithGraphApi()) {
+			$response = $this->graphContext->editUserUsingTheGraphApi($this->getAdminUsername(), $user, null, null, null, null, false);
+		} else {
+			$response = $this->disableOrEnableUser($this->getAdminUsername(), $user, 'disable');
+		}
+		Assert::assertEquals(
+			200,
+			$response->getStatusCode(),
+			" Expected status code is 200 but received " . $response->getStatusCode()
+			. "\nResponse body: " . $response->getBody(),
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/Provisioning.php:3837]:
```suggestion
			__METHOD__ . " Expected status code is 200 but received " . $response->getStatusCode()
			. "\nResponse body: " . $response->getBody(),
```

**Review by saw-jan — APPROVED** [review]
LGTM

**Review by SagarGi — APPROVED** [review]
LGTM :+1:

File: tests/acceptance/features/coreApiShareCreateSpecialToShares2/createShareWithDisabledUser.feature [hunk 9]
```diff
@@ -24,5 +24,4 @@ Feature: share resources with a disabled user
     And user "Alice" has been disabled
```
Comment (SagarGi) [inline tests/acceptance/features/coreApiShareCreateSpecialToShares2/createShareWithDisabledUser.feature:24]: Previously this scenarios seems to be failed. And now why it is failing. Please check it and make new issue if possible.

Comment (SagarGi) [inline tests/acceptance/features/coreApiShareCreateSpecialToShares2/createShareWithDisabledUser.feature:24]: Previously this scenarios seems to be failed. And now why it is failing. Please check it and make new issue if possible.

**Review by SagarGi — CHANGES_REQUESTED** [review]

**Review by SagarGi — APPROVED** [review]

**Pull request merged**